### PR TITLE
refactor: remove uses of `Validation.flatMap` from `Weeder`

### PR DESCRIPTION
**Pull request by mlutze**
related to #3426

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/util/Validation.scala [hunk 1]
```diff
@@ -40,12 +40,15 @@ sealed trait Validation[+T, +E] {
...
+  // Deprecated. Use `andThen` instead.
+  final def flatMap[U, A >: E](f: T => Validation[U, A]): Validation[U, A] = this.andThen(f)
+
   /**
     * Similar to `map` but does not wrap the result in a [[Validation.Success]].
     *
     * Preserves the errors.
     */
-  final def flatMap[U, A >: E](f: T => Validation[U, A]): Validation[U, A] = this match {
+  final def andThen[U, A >: E](f: T => Validation[U, A]): Validation[U, A] = this match {
```
[the remark this answers is not in the document]
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/util/Validation.scala:51]: Renaming it disables Scala's `for { x <- val1, y <- val2 }` sugar for `Validations`, which IMO is too much of a temptation :)

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/util/Validation.scala:51]: Aha, thats a good argument

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/util/Validation.scala:51]: What about calling it flatMapN?

[the remark this answers is not in the document]
Comment (mlutze) [reply main/src/ca/uwaterloo/flix/util/Validation.scala:51]: I'm not strongly opposed to it. My thought was just that it's not a "real" `flatMap`, so the name `andThen` clearly says that we're doing the validation in two separate steps.

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/util/Validation.scala:51]: I am not super sold on this whole thing. But if we rename to `flatMapN` and then use it as a normal function call `flatMapN(....)` then I am OK with merging it in.

**Comment by magnus-madsen** [comment]
Thanks!

**Pull request merged**